src/storage/store/account/onchain_event_store.rs

```diff
@@ -553,6 +555,43 @@ impl OnchainEventStore {
         }
         Ok(None)
     }
+    pub fn is_tier_subscription_active_at(
```

Comment:
I think we'll need to add caching for this, but doesn't have to happen in this PR. Let's add a todo.